mongorestore/mongorestore_test.go

```diff
@@ -2580,3 +2608,122 @@ func testRestoreColumnstoreIndexFromOplog(t *testing.T) {
 		assertColumnstoreIndex(t, testDB, key, columnstoreProjection)
 	})
 }
+
+func TestDumpAndRestoreConfigDB(t *testing.T) {
+	require := require.New(t)
+
+	testtype.SkipUnlessTestType(t, testtype.IntegrationTestType)
+
+	_, err := testutil.GetBareSession()
+	require.NoError(err, "can connect to server")
+
+	t.Run("test dump and restore only config db includes all config collections", func(t *testing.T) {
+		testDumpAndRestoreConfigDBIncludesAllCollections(t)
+	})
+
+	t.Run("test dump and restore all dbs includes only some config collections", func(t *testing.T) {
+		testDumpAndRestoreAllDBsIgnoresSomeConfigCollections(t)
+	})
+}
+
+func testDumpAndRestoreConfigDBIncludesAllCollections(t *testing.T) {
+	require := require.New(t)
+
+	session, err := testutil.GetBareSession()
+	require.NoError(err, "can connect to server")
+
+	configDB := session.Database("config")
+
+	collections := createCollectionsWithTestDocuments(
+		t,
+		configDB,
+		append(configCollectionNamesToKeep, userDefinedConfigCollectionNames...),
+	)
+	defer clearDB(t, configDB)
+
+	withBSONMongodump(
+		t,
+		func(dir string) {
+
+			clearDB(t, configDB)
+
+			restore, err := getRestoreWithArgs(dir)
+			require.NoError(err)
+			defer restore.Close()
+
+			result := restore.Restore()
+			require.NoError(result.Err, "can run mongorestore")
+			require.EqualValues(0, result.Failures, "mongorestore reports 0 failures")
+
+			for _, collection := range collections {
+				r := collection.FindOne(context.Background(), testDocument)
+				require.NoError(r.Err(), "expected document")
+			}
+
+		},
+		"--db", "config",
+		"--excludeCollection", "transactions",
```

Comment:
I had to exclude transactions here because we get an error message when attempting to clear them out, and we get a duplicate key error on restore when attempting to restore. I'm not sure how else to get around this- any other ideas?